File: Assembly-CSharp/SRPG/BuffEffectParam.cs
// Decompiled with JetBrains decompiler
// Type: SRPG.BuffEffectParam
// Assembly: Assembly-CSharp, Version=1.2.0.0, Culture=neutral, PublicKeyToken=null
// MVID: 9BA76916-D0BD-4DB6-A90B-FE0BCC53E511
// Assembly location: C:\Users\André\Desktop\Assembly-CSharp.dll

using System;

namespace SRPG
{
  public class BuffEffectParam
  {
    public string iname;
    public string job;
    public string buki;
    public string birth;
    public ESex sex;
    public int elem;
    public ESkillCondition cond;
    public OInt rate;
    public OInt turn;
    public EffectCheckTargets chk_target;
    public EffectCheckTimings chk_timing;
    public EAppType mAppType;
    public int mAppMct;
    public EEffRange mEffRange;
    public BuffEffectParam.Buff[] buffs;

    public BuffEffectParam.Buff this[ParamTypes type]
    {
      get
      {
        if (this.buffs != null)
          return Array.Find<BuffEffectParam.Buff>(this.buffs, (Predicate<BuffEffectParam.Buff>) (p => p.type == type));
        return (BuffEffectParam.Buff) null;
      }
    }

    public bool Deserialize(JSON_BuffEffectParam json)
    {
      if (json == null)
        return false;
      this.iname = json.iname;
      this.job = json.job;
      this.buki = json.buki;
      this.birth = json.birth;
      this.sex = (ESex) json.sex;
      this.elem = Convert.ToInt32(json.elem.ToString("d7"), 2);
      this.rate = (OInt) json.rate;
      this.turn = (OInt) json.turn;
      this.chk_target = (EffectCheckTargets) json.chktgt;
      this.chk_timing = (EffectCheckTimings) json.timing;
      this.cond = (ESkillCondition) json.cond;
      this.mAppType = (EAppType) json.app_type;
      this.mAppMct = json.app_mct;
      this.mEffRange = (EEffRange) json.eff_range;
      ParamTypes type1 = (ParamTypes) json.type1;
      ParamTypes type2 = (ParamTypes) json.type2;
      ParamTypes type3 = (ParamTypes) json.type3;
      ParamTypes type4 = (ParamTypes) json.type4;
      ParamTypes type5 = (ParamTypes) json.type5;
      ParamTypes type6 = (ParamTypes) json.type6;
      ParamTypes type7 = (ParamTypes) json.type7;
      ParamTypes type8 = (ParamTypes) json.type8;
      ParamTypes type9 = (ParamTypes) json.type9;
      ParamTypes type10 = (ParamTypes) json.type10;
      int length = 0;
      if (type1 != ParamTypes.None)
        ++length;
      if (type2 != ParamTypes.None)
        ++length;
      if (type3 != ParamTypes.None)
        ++length;
      if (type4 != ParamTypes.None)
        ++length;
      if (type5 != ParamTypes.None)
        ++length;
      if (type6 != ParamTypes.None)
        ++length;
      if (type7 != ParamTypes.None)
        ++length;
      if (type8 != ParamTypes.None)
        ++length;
      if (type9 != ParamTypes.None)
        ++length;
      if (type10 != ParamTypes.None)
        ++length;
      if (length > 0)
      {
        this.buffs = new BuffEffectParam.Buff[length];
        int index = 0;
        if (type1 != ParamTypes.None)
        {
          this.buffs[index] = new BuffEffectParam.Buff();
          this.buffs[index].type = type1;
          this.buffs[index].value_ini = (OInt) json.vini1;
          this.buffs[index].value_max = (OInt) json.vmax1;
          this.buffs[index].calc = (SkillParamCalcTypes) json.calc1;
          ++index;
        }
        if (type2 != ParamTypes.None)
        {
          this.buffs[index] = new BuffEffectParam.Buff();
          this.buffs[index].type = type2;
          this.buffs[index].value_ini = (OInt) json.vini2;
          this.buffs[index].value_max = (OInt) json.vmax2;
          this.buffs[index].calc = (SkillParamCalcTypes) json.calc2;
          ++index;
        }
        if (type3 != ParamTypes.None)
        {
          this.buffs[index] = new BuffEffectParam.Buff();
          this.buffs[index].type = type3;
          this.buffs[index].value_ini = (OInt) json.vini3;
          this.buffs[index].value_max = (OInt) json.vmax3;
          this.buffs[index].calc = (SkillParamCalcTypes) json.calc3;
          ++index;
        }
        if (type4 != ParamTypes.None)
        {
          this.buffs[index] = new BuffEffectParam.Buff();
          this.buffs[index].type = type4;
          this.buffs[index].value_ini = (OInt) json.vini4;
          this.buffs[index].value_max = (OInt) json.vmax4;
          this.buffs[index].calc = (SkillParamCalcTypes) json.calc4;
          ++index;
        }
        if (type5 != ParamTypes.None)
        {
          this.buffs[index] = new BuffEffectParam.Buff();
          this.buffs[index].type = type5;
          this.buffs[index].value_ini = (OInt) json.vini5;
          this.buffs[index].value_max = (OInt) json.vmax5;
          this.buffs[index].calc = (SkillParamCalcTypes) json.calc5;
          ++index;
        }
        if (type6 != ParamTypes.None)
        {
          this.buffs[index] = new BuffEffectParam.Buff();
          this.buffs[index].type = type6;
          this.buffs[index].value_ini = (OInt) json.vini6;
          this.buffs[index].value_max = (OInt) json.vmax6;
          this.buffs[index].calc = (SkillParamCalcTypes) json.calc6;
          ++index;
        }
        if (type7 != ParamTypes.None)
        {
          this.buffs[index] = new BuffEffectParam.Buff();
          this.buffs[index].type = type7;
          this.buffs[index].value_ini = (OInt) json.vini7;
          this.buffs[index].value_max = (OInt) json.vmax7;
          this.buffs[index].calc = (SkillParamCalcTypes) json.calc7;
          ++index;
        }
        if (type8 != ParamTypes.None)
        {
          this.buffs[index] = new BuffEffectParam.Buff();
          this.buffs[index].type = type8;
          this.buffs[index].value_ini = (OInt) json.vini8;
          this.buffs[index].value_max = (OInt) json.vmax8;
          this.buffs[index].calc = (SkillParamCalcTypes) json.calc8;
          ++index;
        }
        if (type9 != ParamTypes.None)
        {
          this.buffs[index] = new BuffEffectParam.Buff();
          this.buffs[index].type = type9;
          this.buffs[index].value_ini = (OInt) json.vini9;
          this.buffs[index].value_max = (OInt) json.vmax9;
          this.buffs[index].calc = (SkillParamCalcTypes) json.calc9;
          ++index;
        }
        if (type10 != ParamTypes.None)
        {
          this.buffs[index] = new BuffEffectParam.Buff();
          this.buffs[index].type = type10;
          this.buffs[index].value_ini = (OInt) json.vini10;
          this.buffs[index].value_max = (OInt) json.vmax10;
          this.buffs[index].calc = (SkillParamCalcTypes) json.calc10;
          int num = index + 1;
        }
      }
      return true;
    }

    public class Buff
    {
      public ParamTypes type;
      public OInt value_ini;
      public OInt value_max;
      public SkillParamCalcTypes calc;
    }
  }
}

File: Database_V2/ParamFunctions/BuffEffectParam.py
from ParamFunctions._variables import RAWBIRTH,RAWELEMENT,ENUM,TRANSLATION
def BuffEffectParam(json):
    this={}#BuffEffectParamjson)
    if 'iname' in json:
        this['iname'] = json['iname']
    if 'job' in json:
        this['job'] = json['job']
    if 'buki' in json:
        this['buki'] = json['buki']
    if 'birth' in json:
        this['birth'] = RAWBIRTH[json['birth']]
    if 'sex' in json:
        this['sex'] = ENUM['ESex'][json['sex']]
    if 'un_group' in json:
        this['un_group'] = json['un_group']
    if 'elem' in json:
        this['elem'] = RAWELEMENT[json['elem']]
    if 'rate' in json:
        this['rate'] = json['rate']
    if 'turn' in json:
        this['turn'] = json['turn']
    if 'chktgt' in json:
        this['chk_target'] = ENUM['EffectCheckTargets'][json['chktgt']]
    if 'timing' in json:
        this['chk_timing'] = ENUM['EffectCheckTimings'][json['timing']]
    if 'cond' in json:
        this['cond'] = ENUM['ESkillCondition'][json['cond']]
    if 'up_timing' in json:
        this['mUpTiming'] = ENUM['EffectCheckTimings'][json['up_timing']]
    if 'app_type' in json:
        this['mAppType'] = ENUM['EAppType'][json['app_type']]
    if 'app_mct' in json:
        this['mAppMct'] = json['app_mct']
    if 'eff_range' in json:
        this['mEffRange'] = ENUM['EEffRange'][json['eff_range']]

    this['buffs']=[]
    keys = {
        'type':'type',
        'vini':'value_ini',
        'vmax':'value_max', 
        'vone':'value_one',
        'calc':'calc'
        }
    for i in range(1, 12):
        n=str(i)
        #Fix for false naming
        for k in keys:
            if ('{key}0{i}'.format(key=k,i=n)) in json:
                json[k+n] = json[k+'0'+n]

        if 'type'+n not in json or json['type'+n]==0:
            continue
        if 'vini'+n not in json:
            json['vini'+n]=0

        if 'vmax'+n not in json:
            json['vmax'+n]=0

        if json['vmax'+n] == 0 and json['vini'+n] == 0:
            continue

        this['buffs'].append({
            'type': TRANSLATION[ENUM['ParamTypes'][json['type'+n]]],
            'value_ini': json['vini'+n],
            'value_max': json['vmax'+n],
            'value_one': json['vone'+n] if 'vone'+n in json else 0,
            'calc':ENUM['SkillParamCalcTypes'][json['calc'+n]] if 'calc'+n in json else 'Add'
        })

    this['mIsUpBuff']=False
    for buff in this['buffs']:
        if buff['value_one'] !=0:
            this['mIsUpBuff']=True
            break


    this['mFlags'] = []
    if 'is_up_rep' in json and (json['is_up_rep'] != 0):
        this['mFlags'].append('UpReplenish')
    if 'is_no_dis' in json and (json['is_no_dis'] != 0):
        this['mFlags'].append('NoDisabled')
    if 'is_no_bt' in json and (json['is_no_bt'] != 0):
        this['mFlags'].append('NoBuffTurn')

    if 'custom_targets' in json:
        this['custom_targets'] = json['custom_targets']
    #returntrue
    return this
